docs/developer/update-v1alpha1-provider-to-v1alpha2-infra-provider.md

```diff
@@ -0,0 +1,188 @@
+# Updating a v1alpha1 provider to a v1alpha2 infrastructure provider
+
+This document outlines how to update a cluster API (CAPI) v1alpha1 provider to a v1alpha2 infrastructure provider.
+
+* [Updating a v1alpha1 provider to a v1alpha2 infrastructure provider](#updating-a-v1alpha1-provider-to-a-v1alpha2-infrastructure-provider)
+  * [General information](#general-information)
+    * [The new API group](#the-new-api-group)
+    * [The infrastructure API group](#the-infrastructure-api-group)
+    * [Kubebuilder](#kubebuilder)
+    * [Sample code and other examples](#sample-code-and-other-examples)
+  * [Create a branch for new v1alpha1 work](#create-a-branch-for-new-v1alpha1-work)
+  * [Update the API group in the `PROJECT` file](#update-the-api-group-in-the-project-file)
+  * [Create the provider's v1alpha2 resources](#create-the-providers-v1alpha2-resources)
+    * [The cluster and machine resources](#the-cluster-and-machine-resources)
+    * [The spec and status types](#the-spec-and-status-types)
+      * [Provider cluster status fields](#provider-cluster-status-fields)
+        * [Provider cluster status `ready`](#provider-cluster-status-ready)
+        * [Provider cluster status `apiEndpoints`](#provider-cluster-status-apiendpoints)
+  * [Create the infrastructure controllers](#create-the-infrastructure-controllers)
+    * [The provider cluster controller](#the-provider-cluster-controller)
+    * [The provider machine controller](#the-provider-machine-controller)
+
+## General information
+
+This section contains several general notes about the update process.
+
+### The new API group
+
+Cluster API v1alpha2 custom resource definitions (CRDs) belong to the API group `cluster.x-k8s.io`.
+
+### The infrastructure API group
+
+An infrastructure provider's CRDs should belong to the API group `infrastructure.cluster.x-k8s.io`.
```

Comment:
This should probably also include a caveat that the `infrastructure.cluster.x-k8s.io` api group is only applicable for infrastructure providers that are housed under the `kubernetes-sigs` GitHub org.

Externally owned providers should use an appropriate API group for their ownership and would require additional RBAC rules to be configured and deployed for the common cluster-api components.